### ISSUE TITLE
example not working

### ISSUE THREAD

**Issue by boriskaus** [JuliaParallel/PETSc.jl#109]
Hi,

having an interface to PETSc from Julia would be extremely helpful (we have used PETSc quite a bit in the past). 
Doing this with a BinaryBuilder, which makes it easy to install it on various OS is also an important contribution (would help a lot with teaching, for example).

Yet, whereas the single-core tests work (using Julia 1.6), the single example file that is provided fails, because it seems the `PETSc.C.xxx` no longer exists. 

```
julia> include("PS6_1.jl")
tmax = 4.0
delta_x = 0.02
delta_t = 0.02
r = 50.0
sigma = 1.0
nStep = 199
ERROR: LoadError: UndefVarError: C not defined
Stacktrace:
 [1] createPetscData(mat_size::Int64, stencil_size::Int64)
   @ Main ~/.julia/dev/PETSc/examples/PS6_1.jl:271
 [2] solve(xmin::Int64, xmax::Int64, tmax::Float64, N::Int64, delta_t::Float64)
   @ Main ~/.julia/dev/PETSc/examples/PS6_1.jl:71
 [3] driver()
   @ Main ~/.julia/dev/PETSc/examples/PS6_1.jl:18
 [4] top-level scope
   @ ~/.julia/dev/PETSc/examples/PS6_1.jl:337
 [5] include(fname::String)
   @ Base.MainInclude ./client.jl:444
 [6] top-level scope
   @ REPL[70]:1
in expression starting at /Users/kausb/.julia/dev/PETSc/examples/PS6_1.jl:337
```

It would be great if comment on how this can be fixed.

**Comment by jkozdon** [comment]
Many of the examples and tests are very broken.

From what I can tell the only stuff that works is:
- [test/runtests.jl](https://github.com/JuliaParallel/PETSc.jl/blob/6f68f083431afb52cd3d369ad1b35bcbedd0598a/test/runtests.jl)
- [test/laplacian.jl](https://github.com/JuliaParallel/PETSc.jl/blob/6f68f083431afb52cd3d369ad1b35bcbedd0598a/test/laplacian.jl)
- [examples/SNES_ex2b.jl](https://github.com/JuliaParallel/PETSc.jl/blob/6f68f083431afb52cd3d369ad1b35bcbedd0598a/examples/SNES_ex2b.jl)
- [examples/SNES_ex2.jl](https://github.com/JuliaParallel/PETSc.jl/blob/6f68f083431afb52cd3d369ad1b35bcbedd0598a/examples/SNES_ex2.jl)

I think the ideal would be to make a minimal interface from something like https://github.com/gridap/GridapPETSc.jl which libraries could build on top of. Perhaps merging with the efforts of https://github.com/bmxam/PetscWrap.jl?

Docs are very out of date as well.

**Comment by boriskaus** [comment]
I fully agree. 

We have started working on porting the DMSTAG functionality of PETSc to this package & hope to be able to do a PR of that soon (see [https://github.com/boriskaus/PETSc.jl/tree/feature-DMSTAG](https://github.com/boriskaus/PETSc.jl/tree/feature-DMSTAG)).

PETSc.jl has a number of nice features, including the ability to call Julia routines from PETSc (within SNES) and the fact that it integrates with PETSc_jll (removing the pain of having to install PETSc yourself, which can be a big hurdle). We found it to result in significantly shorter and more readable codes than their C counterparts. Plus you get the ability to incorporate Julia features such as doing automatic differentiation.
 
Yet, documentation, examples and README are mostly out of date. In my opinion, it would be good to extend this to include:

- The ability to link it with an externally compiled PETSc installation (so we can run it on HPC systems where this is installed already).
- Add parallel support back (used to be there, but was removed with the introduction of PETSc_jll; would be g
- Have working examples and move the non-working ones to a different directory (or get rid of them)
- Updated the docs.
